File: memory-domain/src/main/java/com/memory/dto/file/response/FileResponse.java
package com.memory.dto.file.response;

import com.memory.domain.file.File;
import com.memory.domain.file.FileType;

import java.time.LocalDateTime;

public record FileResponse(
        Long id,
        String originalFileName,
        String fileName,
        String fileUrl,
        FileType fileType,
        Long fileSize,
        Long memoryId,
        Long memberId,
        LocalDateTime createDate
) {

    public static FileResponse from(File file) {
        return new FileResponse(
                file.getId(),
                file.getOriginalFileName(),
                file.getFileName(),
                file.getFileUrl(),
                file.getFileType(),
                file.getFileSize(),
                file.getMemory() != null ? file.getMemory().getId() : null,
                file.getMember() != null ? file.getMember().getId() : null,
                file.getCreateDate()
        );
    }
}

File: memory-api/src/main/kotlin/com/memory/dto/file/response/FileResponse.kt
package com.memory.dto.file.response

import com.memory.domain.file.File
import com.memory.domain.file.FileType
import java.time.LocalDateTime

data class FileResponse(
    val id: Long?,
    val originalFileName: String,
    val fileName: String,
    val fileUrl: String,
    val fileType: FileType,
    val fileSize: Long,
    val memoryId: Long?,
    val memberId: Long?,
    val createDate: LocalDateTime?
) {
    companion object {
        @JvmStatic
        fun from(file: File): FileResponse =
            FileResponse(
                id = file.id,
                originalFileName = file.originalFileName,
                fileName = file.fileName,
                fileUrl = file.fileUrl,
                fileType = file.fileType,
                fileSize = file.fileSize,
                memoryId = file.memory?.id,
                memberId = file.member?.id,
                createDate = file.createDate
            )
    }
}